cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/queue/Job.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright © 2017 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.internal.app.runtime.schedule.queue;
+
+import co.cask.cdap.internal.app.runtime.schedule.ProgramSchedule;
+import co.cask.cdap.proto.Notification;
+
+import java.util.List;
+
+/**
+ * A scheduled job.
+ */
+public interface Job {
```

Comment:
I don't think the Job interface needs to have that. It's internally handled using `timestamp` (internal to the JobQueueDataset).